### ISSUE TITLE
How to convert from rotation matrix representation (Bx3x3) to axis-angle representation (Bx3)

### ISSUE THREAD

**Issue by sdxass** [Arthur151/CenterHMR#3]
Hi, thanks for the great work.

Could you please hint me how to convert from rotation matrix representation (Bx3x3) to axis-angle representation (Bx3) (used in original SMPL)?

Thank you!

**Comment by seonghyunkim1212** [comment]
This is the code that converts rotation matrix to axis-angle representation.

def rotmat_to_angleaxis(init_pred_rotmat):

    pred_rotmat_hom = torch.cat([ init_pred_rotmat.reshape(-1, 3, 3), torch.tensor([0,0,1], dtype=torch.float32,).reshape(1, 3, 1).expand(24, -1, -1)], dim=-1)    
    pred_aa = torchgeometry.rotation_matrix_to_angle_axis(pred_rotmat_hom).contiguous().view(1, -1) 
    pred_aa[torch.isnan(pred_aa)] = 0.0   
    pred_aa = pred_aa.view(1,24,3) 

    return pred_aa
